File: src/my_project/medium_problems/from51to100/remove_duplicates_from_sorted_list_ii.py
from typing import List, Optional

# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        
        
        if (head == None):
            return None 
        if (head.next == None):
            return head 
        if (head.val == head.next.val):
            temp = self.deleteDuplicates(head.next)
            while(temp!=None):
                if(temp.val!=head.val):
                    break 
                temp=temp.next
            return temp 
        
        else: 
            head.next = self.deleteDuplicates(head.next)
            return head
                
                
        
        
     

File: src/my_project_typescript/medium_problems_typescript/from1to50/removeDuplicatesFromSortedListII.ts

 
 class ListNode {
     val: number
     next: ListNode | null
      constructor(val?: number, next?: ListNode | null) {
         this.val = (val===undefined ? 0 : val)
          this.next = (next===undefined ? null : next)
      }
 }
 

 function deleteDuplicates(head: ListNode | null): ListNode | null {

    if (head == null){
        return null
    }
    if (head.next == null){
        return head
    }
    if (head.val == head.next.val){
        let temp = deleteDuplicates(head.next)
        while(temp!=null){
            if (temp.val!=head.val){
                break
            }
            temp = temp.next
        }
        return temp
    }
    else{
        head.next = deleteDuplicates(head.next)
        return head
    }

};